app/services/tag_labeling_service.py

```
@@ -0,0 +1,80 @@
+            {"role": "system", "content": system_prompt}
+        ]
+
+        llm_resp = ollama.chat(model="llama3:8b", messages=messages_for_llm)
+        try:
+            response_text = llm_resp["message"]["content"]
+            print(f"LLM response for subject_id {subject_id}:", response_text)
+
+            tag_codes = self.extract_tag_codes_array_from_json_str(response_text)
+
+        except Exception as e:
+            raise Exception(f"LLM response parsing failed for subject_id {subject_id}: " + str(e))
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)